std/algorithm/searching.d

```diff
@@ -4027,6 +4027,64 @@ if (is(typeof(binaryFun!pred(r.front, e))) && isInputRange!R)
 }
 
 /**
+ * An overload of `skipOver` which takes a predicate. Skips over any elements
+ * from the start of the range that evaluate to `true`. Stops popping elements
+ * on the first element which `pred(r.front)` equals `false`.
+ *
+ * Params:
+ *     pred = the test for each element
+ *     r = the range to skip over
+ * Returns:
+ *     `true` if elements were skipped over, `false` otherwise.
+ */
+bool skipOver(alias pred, R)(ref R r) if (ifTestable!(typeof(r.front), unaryFun!pred))
+{
+    bool ret = false;
```

Comment:
That is bad advice.

I can't find a quote at the moment, but Walter has been always saying that default initialization is not meant to provide useful default values, but to make using un-initialized variables deterministic and, wherever possible, an error. When possible, the default value of a type is an invalid one (NaN for floating-point types and '\xFF' for chars, which is an invalid UTF-8 code unit); the reason why most other types are initialized to zero is that they do not have an "invalid" value.

So, do initialize your variables explicitly and do not rely on their default value.